pysph/examples/flow_past_cylinder_2d.py

```diff
@@ -0,0 +1,323 @@
+        x, y = (np.ravel(t) for t in (x, y))
+        x += l_tunnel
+        one = np.ones_like(x)
+        volume = dx * dx * one
+        m = volume * rho
+        outlet = get_particle_array(
+            name='outlet', x=x, y=y, m=m, h=h0, V=1.0/volume, u=umax,
+            uhat=umax, p=0.0, rho=one * rho
+        )
+        return outlet
+
+    def _create_inlet(self):
+        dx = self.dx
+        h0 = self.hdx * self.dx
+        x, y = np.mgrid[dx / 2:n_inlet*dx:dx, -w_tunnel + dx/2:w_tunnel:dx]
+        x, y = (np.ravel(t) for t in (x, y))
+        x = x - n_inlet * dx
+        one = np.ones_like(x)
+        volume = one * dx * dx
+
+        inlet = get_particle_array(
+            name='inlet', x=x, y=y, m=volume * rho, h=h0, u=umax, rho=rho,
+            V=1.0/volume, p=0.0
+        )
+        return inlet
+
+    def create_particles(self):
+        fluid = self._create_fluid()
+        solid = self._create_solid()
+        outlet = self._create_outlet()
+        inlet = self._create_inlet()
+        G.remove_overlap_particles(fluid, solid, dx_solid=self.dx, dim=2)
+
+        particles = [
+            fluid, inlet, outlet, solid
+        ]
+        self.scheme.setup_properties(particles)
+
+        return particles
+
+    def create_scheme(self):
+        '''Other scheme can be added here'''
+        h0 = nu = None
+        self.iom = self._create_inlet_outlet_manager()
+
+        edac = EDACScheme(
+            ['fluid'], ['solid'], dim=2, rho0=rho, c0=c0, h=h0,
+            pb=None, nu=nu, alpha=0.2, inlet_outlet_manager=self.iom
+        )
+
+        s = SchemeChooser(default='edac', edac=edac)
+
+        return s
+
+    def configure_scheme(self):
+        scheme = self.scheme
+        h0 = self.hdx * self.dx
+        pfreq = 100
+        kernel = QuinticSpline(dim=2)
+        self.iom.update_dx(self.dx)
+        if self.options.scheme == 'edac':
+            scheme.configure(h=h0, nu=self.nu, pb=p0)
+
+        scheme.configure_solver(kernel=kernel, tf=self.tf, dt=self.dt,
+                                pfreq=pfreq, n_damp=0)
+
+    def _create_inlet_outlet_manager(self):
+        inleteqns = [
+            ResetInletVelocity('inlet', [], U=umax, V=0.0, W=0.0),
+            SolidWallPressureBC('inlet', ['fluid'])
+        ]
+
+        inlet_info = InletInfo(
+            pa_name='inlet', normal=[-1.0, 0.0, 0.0],
+            refpoint=[0.0, 0.0, 0.0], equations=inleteqns
+        )
+
+        outlet_info = OutletInfo(
+            pa_name='outlet', normal=[1.0, 0.0, 0.0],
+            refpoint=[l_tunnel, 0.0, 0.0]
+        )
+
+        iom = SimpleInletOutlet(
+            fluid_arrays=['fluid'], inletinfo=[inlet_info],
+            outletinfo=[outlet_info]
+        )
+
+        return iom
+
+    def create_inlet_outlet(self, particle_arrays):
+        iom = self.iom
+        io = iom.get_inlet_outlet(particle_arrays)
+        iom.update_inlet_outlet_domain()
```

Comment:
Can this be automatically called in `get_inlet_outlet`? I would imagine a user would forget to do this.

Reply:
removed it. as I was using it for removing particles when It left the domain not from outlet.